toolkit/tools/pkg/imagecustomizerlib/liveosisobuilder.go

```diff
@@ -231,66 +244,109 @@ func (b *LiveOSIsoBuilder) prepareRootfsForDracut(writeableRootfsDir string) err
+		if err != nil {
+			return fmt.Errorf("failed to update the kernel file path in the iso grub.cfg:\n%w", err)
+		}
+
+		inputContentString, err = setInitrdPathAll(inputContentString, isoInitrdPath)
+		if err != nil {
+			return fmt.Errorf("failed to update the initrd file path in the iso grub.cfg:\n%w", err)
+		}
 	}
 
-	inputContentString, err = replaceToken(inputContentString, oldLinuxPath, isoKernelPath)
+	rootValue := fmt.Sprintf(rootValueTemplate, isomakerlib.DefaultVolumeId)
+	inputContentString, err = replaceKernelCommandLineArgValueAll(inputContentString, "root", rootValue)
 	if err != nil {
-		return fmt.Errorf("failed to update all the kernel file path occurances in the iso grub.cfg:\n%w", err)
+		return fmt.Errorf("failed to update the root kernel argument in the iso grub.cfg:\n%w", err)
 	}
 
-	inputContentString, oldInitrdPath, err := setInitrdPath(inputContentString, isoInitrdPath)
+	inputContentString, err = updateSELinuxCommandLineHelperAll(inputContentString, imagecustomizerapi.SELinuxModeDisabled)
 	if err != nil {
-		return fmt.Errorf("failed to update the initrd file path in the iso grub.cfg:\n%w", err)
+		return fmt.Errorf("failed to set SELinux mode:\n%w", err)
 	}
 
-	inputContentString, err = replaceToken(inputContentString, oldInitrdPath, isoInitrdPath)
+	mergedExtraCommandLine, err := mergeKernelParameters(savedConfigKernelArgsFilePath, extraCommandLine)
 	if err != nil {
-		return fmt.Errorf("failed to update all the initrd file path occurances in the iso grub.cfg:\n%w", err)
+		return fmt.Errorf("failed to combine additional kernel command line parameters:\n%w", err)
 	}
 
-	inputContentString, _, err = replaceKernelCommandLineArgValue(inputContentString, "root", rootValue)
+	liveosKernelArgs := fmt.Sprintf(kernelArgsLiveOSTemplate, liveOSDir, liveOSImage)
+	additionalKernelCommandline := " " + kernelArgsConsoleTemplate + " " + liveosKernelArgs + " " + mergedExtraCommandLine
+
+	inputContentString, err = appendKernelCommandLineArgsAll(inputContentString, additionalKernelCommandline)
 	if err != nil {
-		return fmt.Errorf("failed to update the root kernel argument in the iso grub.cfg:\n%w", err)
+		return fmt.Errorf("failed to update the kernel arguments with the LiveOS configuration and user configuration in the iso grub.cfg:\n%w", err)
 	}
 
-	inputContentString, err = updateSELinuxCommandLineHelper(inputContentString, imagecustomizerapi.SELinuxModeDisabled)
+	err = os.MkdirAll(filepath.Dir(savedConfigKernelArgsFilePath), os.ModePerm)
 	if err != nil {
-		return fmt.Errorf("failed to set SELinux mode:\n%w", err)
+		return fmt.Errorf("failed to create directory for (%s):\n%w", savedConfigKernelArgsFilePath, err)
 	}
-
-	liveosKernelArgs := fmt.Sprintf(kernelArgsTemplate, liveOSDir, liveOSImage, extraCommandLine)
-
-	inputContentString, err = appendKernelCommandLineArgs(inputContentString, liveosKernelArgs)
+	err = os.WriteFile(savedConfigKernelArgsFilePath, []byte(mergedExtraCommandLine), 0o644)
```

Comment:
nit: Use `file.Write`. (Just for the extra logging.)

Reply:
pending.

Reply:
fixed.